ports/rp2/machine_adc.c

```diff
@@ -72,18 +82,26 @@ STATIC mp_obj_t machine_adc_make_new(const mp_obj_type_t *type, size_t n_args, s
 
     } else {
         // Get GPIO and check it has ADC capabilities.
-        channel = mp_hal_get_pin_obj(source);
-        if (!ADC_IS_VALID_GPIO(channel)) {
+        const machine_pin_obj_t *pin = machine_pin_find(source);
+        channel = pin->id;
+        #if MICROPY_HW_ADC_EXT_COUNT
+        is_ext = pin->is_ext;
+        #endif
+        if (!(!is_ext && ADC_IS_VALID_GPIO(channel)) && !(is_ext && machine_pin_ext_is_adc(pin))) {
```

Comment:
This logic is pretty hard to read now.  I suggest changing to something like this:
```c
bool valid_adc_pin = false;
#if MICROPY_HW_ADC_EXT_COUNT
is_ext = pin->is_ext;
if (is_ext) {
    if (machine_pin_ext_is_adc(pin)) {
        valid_adc_pin = true;
    }
} else
#endif
{
    if (ADC_IS_VALID_GPIO(channel)) {
        valid_adc_pin = true;
    }
}
if (!valid_adc_pin) {
    mp_raise_ValueError(...);
}
```

Reply:
Okay